comp/core/autodiscovery/scheduler/controller.go

```diff
@@ -0,0 +1,204 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package scheduler
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+// MaxRetries is the maximum number of retries for a failed task (TODO: implement retries on failure)
+// MaxRetries = 5
+)
+
+// Controller is a scheduler dispatching to all its registered schedulers
+type Controller struct {
+	// m protects all fields in this struct.
+	m sync.Mutex
+
+	// activeSchedulers is the set of schedulers currently subscribed to configs.
+	activeSchedulers map[string]Scheduler
+
+	// scheduledConfigs contains the set of configs that have been scheduled
+	// via the schedulerController, but not subsequently unscheduled.
+	scheduledConfigs map[string]*integration.Config
+
+	// ConfigStateStore contains the desired state of configs
+	ConfigStateStore *ConfigStateStore
```

Comment:
I see this field is public and a direct call:
```
digests := ac.schedulerController.ConfigStateStore.UpdateDesiredState(changes)
```
I believe it should be private and `ApplyChanges` should be changed to take a list of configs and perform the call to `UpdateDesiredState(changes)`

Reply:
Changed to private.